controllers/servers/host_definer/watcher/host_definition_watcher.py

```diff
@@ -57,19 +57,27 @@ def _is_host_definition_in_desired_state(self, host_definition):
     def _handle_pending_host_definition(self, host_definition):
         response = DefineHostResponse()
         phase = host_definition.phase
+        action = self._get_action(phase)
         if phase == settings.PENDING_CREATION_PHASE:
             response = self._define_host(host_definition)
         elif self._is_pending_for_deletion_need_to_be_handled(phase, host_definition.node_name):
             response = self._undefine_host(host_definition)
         self._handle_error_message_for_pending_host_definition(
-            host_definition, response.error_message)
+            host_definition, response.error_message, action)
 
-    def _handle_error_message_for_pending_host_definition(self, host_definition, error_message):
+    def _get_action(self, phase):
+        if phase == settings.PENDING_CREATION_PHASE:
+            return settings.DEFINE
+        elif phase == settings.PENDING_DELETION_PHASE:
+            return settings.UNDEFINE
+        return ''
+
+    def _handle_error_message_for_pending_host_definition(self, host_definition, error_message, action):
         phase = host_definition.phase
         if error_message:
-            self._add_event_to_host_definition(host_definition, str(error_message))
+            self._create_fail_event_for_host_definition(host_definition, str(error_message), action)
         elif phase == settings.PENDING_CREATION_PHASE:
-            self._set_host_definition_status(host_definition.name, settings.READY_PHASE)
+            self._set_host_definition_status_to_ready(host_definition)
```

Comment:
why do we want to send a `success` event here?

Reply:
we want to send a success event every time that we change the status to Ready, and here we change the status from `PendingCreation` to `Ready`

Reply:
dont we have an error there? its in `_handle_error_message_for_pending_host_definition`

Reply:
not necessarily, I need to change the name of this function, I wanted to do it at CSI-4567 ticket

Reply:
ok, can you add this conversation to the ticket only so we won't forget it?

Reply:
done